test/cmd/newapp.sh

```diff
@@ -206,30 +206,31 @@ os::cmd::expect_failure_and_text "echo 'S P A C E S=test' | oc new-app php --bui
+os::cmd::expect_success_and_text "oc new-build --binary php --to=binary --build-env-file ${build_env_file} -o jsonpath='{.items[?(@.kind==\"BuildConfig\")].spec.strategy.sourceStrategy.env[?(@.name==\"SOME_VAR\")].value}'" 'buildenvvarfromfile'
+os::cmd::expect_success_and_text "oc new-build --binary php --to=binary --build-env-file ${build_env_file} --env SOME_VAR=fromcmdline -o jsonpath='{.items[?(@.kind==\"BuildConfig\")].spec.strategy.sourceStrategy.env[?(@.name==\"SOME_VAR\")].value}'" 'fromcmdline'
+os::cmd::expect_success_and_text "oc new-build --binary php --to=binary --build-env-file ${build_env_file} --env SOME_VAR=fromcmdline -o yaml" 'ignoring value from file'
+os::cmd::expect_success_and_text "cat ${build_env_file} | oc new-build --binary php --to=binary --build-env-file - -o jsonpath='{.items[?(@.kind==\"BuildConfig\")].spec.strategy.sourceStrategy.env[?(@.name==\"SOME_VAR\")].value}'" 'buildenvvarfromfile'
 
 os::cmd::expect_failure_and_text "oc new-build --binary php --build-env-file does/not/exist" 'no such file or directory'
 os::cmd::expect_failure_and_text "oc new-build --binary php --build-env-file test/testdata"  'is a directory'
-os::cmd::expect_success "oc new-build --binary php --build-env-file /dev/null -o yaml"
-os::cmd::expect_success "oc new-build --binary php --build-env-file /dev/null --env-file ${build_env_file} -o yaml"
-os::cmd::expect_failure_and_text "echo 'fo%(o=bar' | oc new-build --binary php --build-env-file -" 'invalid parameter assignment'
-os::cmd::expect_failure_and_text "echo 'S P A C E S=test' | oc new-build --binary php --build-env-file -" 'invalid parameter assignment'
+os::cmd::expect_success "oc new-build --binary php --to=binary --build-env-file /dev/null -o yaml"
+os::cmd::expect_success "oc new-build --binary php --to=binary --build-env-file /dev/null --env-file ${build_env_file} -o yaml"
+os::cmd::expect_failure_and_text "echo 'fo%(o=bar' | oc new-build --binary php --to=binary --build-env-file -" 'invalid parameter assignment'
+os::cmd::expect_failure_and_text "echo 'S P A C E S=test' | oc new-build --binary php --to=binary --build-env-file -" 'invalid parameter assignment'
 
 # new-build - check that we can set build args in DockerStrategy
 os::cmd::expect_success_and_text "oc new-build ${OS_ROOT}/test/testdata/build-arg-dockerfile --build-arg 'foo=bar' --to 'test' -o jsonpath='{.items[?(@.kind==\"BuildConfig\")].spec.strategy.dockerStrategy.buildArgs[?(@.name==\"foo\")].value}'" 'bar'
 
 # check that we cannot set build args in a non-DockerStrategy build
 os::cmd::expect_failure_and_text "oc new-build https://github.com/openshift/ruby-hello-world --strategy=source --build-arg 'foo=bar'" "error: Cannot use '--build-arg' without a Docker build"
+os::cmd::expect_success "oc delete is ruby-ex --ignore-not-found"
```

Comment:
why was this needed?  and why the --ignore-not-found? Our tests should be deterministic so either we should expect this is there to be deleted, or not.

Reply:
Leftover from prior incarnations ... I'll remove ... probably should be fine ...if things break, I'll be able to provide a specific reason why

Reply:
tests pass with this removed .... will be part of next push